### PR TITLE
Add ID Tests

### PR DESCRIPTION
**Pull request by mohitbadve**
The tests are ID tests detected by NonDex.
Java Version - 17.0.13
Maven Version - 3.9.9

Test | Path of Successful When Run without NonDex | Path of Failure When Run with NonDex | Percentage of Failure (nondexRuns=100)
---- | ---------------------------------------- | ---------------------------------- | ---------
org.apache.activemq.jaas.LDAPModuleRoleExpansionTest.testRoleExpansion | /home/mbadve2/newrepos/activemq/activemq-jaas/mvn-1-testRoleExpansion-01-Dec-2024-211500.log  | /home/mbadve2/newrepos/activemq/activemq-jaas/.nondex/yfHB5oLsPKl4e9mQpXh2N8eRU5x1cqcBuahDJiNJM4= | 98
 |  |  | 
org.apache.activemq.jaas.LDAPLoginModuleTest.testAESEncryptedLogin | /home/mbadve2/newrepos/activemq/activemq-jaas/mvn-1-testAESEncryptedLogin-01-Dec-2024-214212.log  | /home/mbadve2/newrepos/activemq/activemq-jaas/.nondex/vrA8gOHcIsMsJsBKHZXPPLUer8CPgDtjAhxUja1w= | 69
 
**VM ID: 044**

### CODE REVIEW

**Comment by darko-marinov** [comment]
Did you run tests individually to confirm them? Can you check how https://github.com/TestingResearchIllinois/idoft/tree/main/verify_test_type can help you confirm the type?

**Comment by mohitbadve** [comment]
Yes, I ran the tests individually. 

The script helps confirm the type surely, logs are here - `/home/mbadve2/newrepos/activemq/testRoleExpansion`,  `/home/mbadve2/newrepos/activemq/testAESEncryptedLogin`

I have tried enhancing it to also change nondex version dynamically based on the current system default java version.

File: verify_test_type/verify_test_type.sh [hunk 1]
```diff
@@ -3,29 +3,57 @@
 # Check if all required arguments are provided
 if [ "$#" -ne 4 ]; then
   echo "Ensure that you are running the command from the repo root"
-  echo "Usage: $0 <module_name> <commit_hash> <test_name> <test_class>"
+  echo "Usage: $0 <commit_hash> <module_name> <test_class> <test_name>"
```
Comment (darko-marinov) [inline verify_test_type/verify_test_type.sh:6]: Why change the order of arguments? Someone else is trying to write `README` (#1555), so you should keep in sync.

Comment (mohitbadve) [reply verify_test_type/verify_test_type.sh]: Oh, I was not aware of the `README` change. I thought to keep the order intuitive like how we have in pr-data.csv. I have reverted it back.

**Comment by darko-marinov** [comment]
@yugvajani can you review these changes to the script that you're documenting?

**Comment by yugvajani** [comment]
> @yugvajani can you review these changes to the script that you're documenting?

The changes are handling the correct nondex version for all the java versions. @mohitbadve any reason for removing the $target from this base_dir="$current_dir/$target/$test_name/logs". Rest of the PR looks good to me.

**Comment by yugvajani** [comment]
@mohitbadve The target is likely intended to represent only the folder, but since it is not explicitly initialized, it might be better to use a different variable for the log names. This is because the current approach could result in similar log names if there are two tests with the same name in different classes.

**Comment by mohitbadve** [comment]
I second that. I updated the folder name to test_class.test_name now

**Comment by mohitbadve** [comment]
@darko-marinov @yugvajani could you please review the PR now

**Comment by yugvajani** [comment]
Looks good

**Comment by darko-marinov** [comment]
Thank you both. Accepting for now. Future students can improve further. It's great to see some other students already using some scripts.

**Pull request merged**